neo/Consensus/ConsensusService.cs

```diff
@@ -206,61 +225,62 @@ private void InitializeConsensus(byte view_number)
 
         private void LocalNode_InventoryReceived(object sender, IInventory inventory)
         {
-            ConsensusPayload payload = inventory as ConsensusPayload;
-            if (payload != null)
+            if (inventory == null || !(inventory is ConsensusPayload payload)) return;
+
+            lock (context)
             {
-                lock (context)
+                if (payload.ValidatorIndex == context.MyIndex ||
+                    payload.ValidatorIndex >= context.Validators.Length ||
+                    payload.Version != ConsensusContext.Version) return;
+
+                if (payload.PrevHash != context.PrevHash || payload.BlockIndex != context.BlockIndex)
                 {
-                    if (payload.ValidatorIndex == context.MyIndex) return;
+                    // Request blocks
 
-                    if (payload.Version != ConsensusContext.Version)
-                        return;
-                    if (payload.PrevHash != context.PrevHash || payload.BlockIndex != context.BlockIndex)
+                    if (Blockchain.Default?.Height + 1 < payload.BlockIndex)
                     {
-                        // Request blocks
+                        Log($"chain sync: expected={payload.BlockIndex} current: {Blockchain.Default?.Height} nodes={localNode.RemoteNodeCount}");
 
-                        if (Blockchain.Default?.Height + 1 < payload.BlockIndex)
-                        {
-                            Log($"chain sync: expected={payload.BlockIndex} current: {Blockchain.Default?.Height} nodes={localNode.RemoteNodeCount}");
+                        localNode.RequestGetBlocks();
+                    }
 
-                            localNode.RequestGetBlocks();
-                        }
+                    return;
+                }
 
-                        return;
-                    }
+                ConsensusMessage message;
+                try
+                {
+                    message = ConsensusMessage.DeserializeFrom(payload.Data);
+                }
+                catch
+                {
+                    return;
+                }
 
-                    if (payload.ValidatorIndex >= context.Validators.Length) return;
```

Comment:
Check if payload validator Index could had changed on RequestGetBlocks()

Reply:
I move this validation to the begining of the method
https://github.com/neo-project/neo/pull/320/files#diff-0285c1c12d1d492897a99ffe07f9fed9R233